lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_localize_helper.rb

```diff
@@ -285,18 +286,15 @@ def self.download_from_glotpress(res_dir:, glotpress_project_url:, locales_map:,
         #
         def self.download_glotpress_export_file(project_url:, locale:, filters:)
           query_params = filters.transform_keys { |k| "filters[#{k}]" }.merge(format: 'android')
-          uri = URI.parse("#{project_url.chomp('/')}/#{locale}/default/export-translations/?#{URI.encode_www_form(query_params)}")
+          url = "#{project_url.chomp('/')}/#{locale}/default/export-translations/?#{URI.encode_www_form(query_params)}"
 
-          # Set an unambiguous User Agent so GlotPress won't rate-limit us
-          options = { 'User-Agent' => Wpmreleasetoolkit::USER_AGENT }
-
-          begin
-            uri.open(options) { |f| Nokogiri::XML(f.read.gsub("\t", '    '), nil, Encoding::UTF_8.to_s) }
-          rescue StandardError => e
-            UI.error "Error downloading #{locale} - #{e.message}"
-            retry if e.is_a?(OpenURI::HTTPError) && UI.confirm("Retry downloading `#{locale}`?")
-            nil
+          downloader = Fastlane::Helper::GlotPressDownloader.new(auto_retry: true)
+          result = nil
+          downloader.download(url, locale) do |response_body|
+            # Replace tabs with spaces (GlotPress uses tabs, but we prefer spaces)
+            result = Nokogiri::XML(response_body.gsub("\t", '    '), nil, Encoding::UTF_8.to_s)
           end
+          result
```

Comment:
Make the `downloader.download` method return the return value of the inner block it calls, so that you can then write this like this, with implicit return of the return value of `downloader.download`:

```suggestion
          downloader.download(url, locale) do |response_body|
            # Replace tabs with spaces (GlotPress uses tabs, but we prefer spaces)
            Nokogiri::XML(response_body.gsub("\t", '    '), nil, Encoding::UTF_8.to_s)
end
```

Reply:
Actually I see you already have implemented a `download_and_return` helper for that in the downloader class… so you probably want to use it here?

Reply:
Done in 901a62c63f678ba55ccde774a987ef41230cf09a.